packages/neos-ui/src/Sagas/UI/Hotkeys/index.js

```diff
@@ -1,20 +1,36 @@
 import Mousetrap from 'mousetrap';
-import {take, race} from 'redux-saga/effects';
-import {actionTypes} from '@neos-project/neos-ui-redux-store';
+import {take, race, put} from 'redux-saga/effects';
+import {actionTypes, actions} from '@neos-project/neos-ui-redux-store';
 import {getGuestFrameDocument} from '@neos-project/neos-ui-guest-frame/src/dom';
 
 export function * handleHotkeys({globalRegistry, store}) {
     yield take(actionTypes.System.READY);
 
-    const hotkeyRegistry = globalRegistry.get('hotkeys');
-    const items = hotkeyRegistry.getAllAsList();
+    const keyboardShortcuts = globalRegistry.get('frontendConfiguration').get('keyboardShortcuts');
     let mousetrapPaused = false;
 
-    items.forEach(item => {
-        Mousetrap.bind(item.keys, () => {
-            store.dispatch(item.action());
+    // We get the action as a string from settings
+    // and this needs to be converted into a real action
+    const getActualAction = actionAsString => actionAsString.split('.').reduce((action, actionStringPart) => {
+        if (action && actionStringPart in action) {
+            return action[actionStringPart];
+        }
+
+        return null;
+    }, actions);
+
+    for (const keyboardShortcut of Object.values(keyboardShortcuts)) {
+        const actualAction = getActualAction(keyboardShortcut.action.substr(keyboardShortcut.action.indexOf('.') + 1));
```

Comment:
I'm not quite sure about coupling keyboard actions to redux actions. Alternatively we could still keep the registry of all actions callable via keyboard. Would be more flexible, but also more boilerplate to write.
Anyways, not opinionated here. @theilm how do you feel about this?

Reply:
I'm not sure whether it would be better to be able to call arbitrary js-functions or only redux-actions.
I mean everything that is meaningful should be represented in the store and so manipulatable via a redux action anyway, or am I wrong?

Reply:
Perhaps you are right.

Reply:
I'm not sure which path we should follow.
Having a registry where we store all available keyboard shortcuts which the 'user' can use would make it a lot safer but will increase our work amount. At the same time, it is more limiting to the user who maybe comes up with ideas we didn't even have considered how to use something.
In a registry, we could define more complex functions which would be hard with only settings.yaml.

So, what are your thought dumps?